pkg/upgrade/base.go

```diff
@@ -132,12 +133,16 @@ func (u *base) UpdateProviderIDsToNodes() error {
 	pairs := make(map[string]*v1.Node)
 	for i := range nodes.Items {
 		node := nodes.Items[i]
-		providerID, err := kubernetes2.ParseProviderID(node.Spec.ProviderID)
-		if err != nil {
+		id := ""
+		providerID, err := noderefutil.NewProviderID(node.Spec.ProviderID)
+		if err == nil {
+			id = providerID.ID()
+		} else {
+			u.log.Error(err, "failed to parse provider id", "id", node.Spec.ProviderID)
 			// unable to parse provider ID with whitelist of provider ID formats. Use original provider ID
-			providerID = node.Spec.ProviderID
+			id = node.Spec.ProviderID
 		}
-		pairs[providerID] = &node
+		pairs[id] = &node
```

Comment:
Can we use `providerID` as the key instead of `id`? That would allow us to compare on both provider (e.g. `aws`) and id, instead of just id.

Reply:
Nevermind 😄